app/commands/exercise/representation/find_example_submissions.rb

```diff
@@ -0,0 +1,44 @@
+class Exercise::Representation::FindExampleSubmissions
+  include Mandate
+
+  initialize_with :representation
+
+  def call
+    example_submissions = [representation.source_submission]
+    example_submission_hashes = Set.new([solution_files_hash(representation.source_submission)])
+    page = 1
+
+    loop do
+      submissions = representation.
+        submission_representation_submissions.
+        where.not(id: representation.source_submission.id).
```

Comment:
For the sake of speed, I suggest removing this constraint in the SQL and moving it into the Ruby.